airbyte-integrations/connectors/source-outreach/source_outreach/source.py

```diff
@@ -47,39 +68,89 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
             raise KeyError(f"error parsing next_page token: {e}")
 
     def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, any] = None,
+        next_page_token: Mapping[str, Any] = None,
     ) -> MutableMapping[str, Any]:
-        params = {"page[size]": 100, "count": "false"}
+        params = {"page[size]": self.page_size, "count": "false"}
         if next_page_token and "after" in next_page_token:
             params["page[after]"] = next_page_token["after"]
+
+        # Really just a place to handle cycling refresh tokens that has access to the state.
+        if (
+            "refresh_token" in stream_state
+            and self.authenticator.cycling_refresh_token is None
+        ):
+            self.authenticator.refresh_token = stream_state["refresh_token"]
+        elif self.authenticator.cycling_refresh_token is not None:
+            stream_state["refresh_token"] = self.authenticator.cycling_refresh_token
+
         return params
 
-    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+    def parse_response(
+        self, response: requests.Response, **kwargs
+    ) -> Iterable[Mapping]:
         data = response.json().get("data")
         if not data:
             return
         for element in data:
-            yield element
+            relationships: Dict[str, List[int]] = dict()
+            for r_type, relations in element.get("relationships").items():
+                relationships[f"{r_type}"] = []
+                if relations.get(
+                    "data"
+                ):  # Manage None and pass empty data. Some relationships only have links we do not handle these.
+                    data = relations.get("data", [])
+
+                    if isinstance(
+                        data, dict
+                    ):  # Manage some relationships that only have one element and are set as dict.
+                        # instead of having [{'type': 'sequenceState', 'id': 1}] we have {'type': 'sequenceState', 'id': 1}
+                        data = [data]
+
+                    relationships[f"{r_type}"] = [e.get("id") for e in data]
+
+            yield {
+                **element.get("attributes"),
+                **{self.primary_key: element[self.primary_key], **relationships},
+            }
 
 
 # Basic incremental stream
 class IncrementalOutreachStream(OutreachStream, ABC):
     @property
     def cursor_field(self) -> str:
-        return "attributes/properties/updatedAt"
+        return "updatedAt"
```

Comment:
Thoughts on naming this `UpdatedAtIncrementalOutreachStream` for consistency?

Reply:
I wanted to, but figured for when merging back to the original. Looking at this implementation this seems like written by a beginner with airbyte integrations. :shrug:  

Current Version naming:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-outreach/source_outreach/source.py#L80